File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/org/ds/model/PayAccountMethodDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.org.ds.model;

import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;

import net.nan21.dnet.module.md.org.domain.entity.PayAccountMethod;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = PayAccountMethod.class)
public class PayAccountMethodDs extends AbstractAuditableDs<PayAccountMethod> {

    public static final String fPAYACCOUNTID = "payAccountId";
    public static final String fPAYACCOUNT = "payAccount";
    public static final String fPAYACCOUNTTYPE = "payAccountType";
    public static final String fPAYMETHODID = "payMethodId";
    public static final String fPAYMETHOD = "payMethod";
    public static final String fALLOWPAYIN = "allowPayIn";
    public static final String fALLOWPAYOUT = "allowPayOut";

    @DsField(join = "left", path = "payAccount.id")
    private Long payAccountId;

    @DsField(join = "left", path = "payAccount.name")
    private String payAccount;

    @DsField(join = "left", path = "payAccount.type")
    private String payAccountType;

    @DsField(join = "left", path = "payMethod.id")
    private Long payMethodId;

    @DsField(join = "left", path = "payMethod.name")
    private String payMethod;

    @DsField()
    private Boolean allowPayIn;

    @DsField()
    private Boolean allowPayOut;

    public PayAccountMethodDs() {
        super();
    }

    public PayAccountMethodDs(PayAccountMethod e) {
        super(e);
    }

    public Long getPayAccountId() {
        return this.payAccountId;
    }

    public void setPayAccountId(Long payAccountId) {
        this.payAccountId = payAccountId;
    }

    public String getPayAccount() {
        return this.payAccount;
    }

    public void setPayAccount(String payAccount) {
        this.payAccount = payAccount;
    }

    public String getPayAccountType() {
        return this.payAccountType;
    }

    public void setPayAccountType(String payAccountType) {
        this.payAccountType = payAccountType;
    }

    public Long getPayMethodId() {
        return this.payMethodId;
    }

    public void setPayMethodId(Long payMethodId) {
        this.payMethodId = payMethodId;
    }

    public String getPayMethod() {
        return this.payMethod;
    }

    public void setPayMethod(String payMethod) {
        this.payMethod = payMethod;
    }

    public Boolean getAllowPayIn() {
        return this.allowPayIn;
    }

    public void setAllowPayIn(Boolean allowPayIn) {
        this.allowPayIn = allowPayIn;
    }

    public Boolean getAllowPayOut() {
        return this.allowPayOut;
    }

    public void setAllowPayOut(Boolean allowPayOut) {
        this.allowPayOut = allowPayOut;
    }

}

File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/PayAccountMethodDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.md.org.ds.model.PayAccountMethodDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"payAccountId", type:"int", useNull:true},
		{name:"payAccount", type:"string"},
		{name:"payAccountType", type:"string"},
		{name:"payMethodId", type:"int", useNull:true},
		{name:"payMethod", type:"string"},
		{name:"allowPayIn", type:"boolean"},
		{name:"allowPayOut", type:"boolean"}
	]
});

Ext.define("net.nan21.dnet.module.md.org.ds.model.PayAccountMethodDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"payAccountId", type:"int", useNull:true},
		{name:"payAccountId_From",type:"int", useNull:true},
		{name:"payAccountId_To",type:"int", useNull:true},
		{name:"payAccount", type:"string"},
		{name:"payAccountType", type:"string"},
		{name:"payMethodId", type:"int", useNull:true},
		{name:"payMethodId_From",type:"int", useNull:true},
		{name:"payMethodId_To",type:"int", useNull:true},
		{name:"payMethod", type:"string"},
		{name:"allowPayIn", type:"boolean", useNull:true},
		{name:"allowPayOut", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.md.org.ds.param.PayAccountMethodDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 